File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraCharts;

namespace ChartStackedSorting
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            chartControl1.Series.Clear();

            Series s1, s2, s3;

            s1 = new Series("Serie1", ViewType.StackedBar);
            s2 = new Series("Serie2", ViewType.StackedBar);
            s3 = new Series("Serie3", ViewType.StackedBar);

            Random r = new Random();

            for(int i = 0; i < 10; i++) {
                s1.Points.Add(new SeriesPoint(i, Math.Round(r.NextDouble() * 100)));
                s2.Points.Add(new SeriesPoint(i, Math.Round(r.NextDouble() * 100)));
                s3.Points.Add(new SeriesPoint(i, Math.Round(r.NextDouble() * 100)));
            }

            s1.Label.Visible = s2.Label.Visible = s3.Label.Visible = false;

            chartControl1.Series.AddRange(new Series[] { s1, s2, s3 });
        }

        private void button1_Click(object sender, EventArgs e) {
            ChartStackedSorting css = new ChartStackedSorting(chartControl1);

            css.SortChartSeries();
        }
    }

  public class ChartStackedSorting {
    private ChartControl chart;

    public ChartStackedSorting(ChartControl chart) {
        this.chart = chart;
    }

    public void SortChartSeries() {
        for(int i = 0; i < chart.Series[0].Points.Count; i++) {
            SeriesPoint p1 = chart.Series[0].Points[i];
            SeriesPoint pointMax = p1;
            int indexMax = i;

            double secondItemsMax = GetAdditionalValues(p1);

            for(int j = i + 1; j < chart.Series[0].Points.Count; j++) {
                SeriesPoint p2 = chart.Series[0].Points[j];

                double secondItems = GetAdditionalValues(p2);

                if(p2.Values[0] + secondItems > pointMax.Values[0] + secondItemsMax) {
                    pointMax = p2;
                    secondItemsMax = secondItems;
                    indexMax = j;
                }
            }

            chart.Series[0].Points.Swap(p1, pointMax);
            SwapAdditionalPoints(p1, pointMax);
        }
    }

    private double GetAdditionalValues(SeriesPoint point) {
        double result = 0.0;

        for(int i = 1; i < chart.Series.Count; i++)
            for(int j = 0; j < chart.Series[i].Points.Count; j++)
                if(Convert.ToDouble(chart.Series[i].Points[j].Argument) == Convert.ToDouble(point.Argument) && chart.Series[i].Points[j] != point)
                    result += chart.Series[i].Points[j].Values[0];

        return result;
    }

    private void SwapAdditionalPoints(SeriesPoint pi, SeriesPoint pj) {
        for(int s = 1; s < chart.Series.Count; s++) {

            SeriesPoint swp1 = null;
            SeriesPoint swp2 = null;
            
            for(int i = 0; i < chart.Series[s].Points.Count; i++) {
                if(Convert.ToDouble(chart.Series[s].Points[i].Argument) == Convert.ToDouble(pi.Argument))
                    swp1 = chart.Series[s].Points[i];
                if(Convert.ToDouble(chart.Series[s].Points[i].Argument) == Convert.ToDouble(pj.Argument))
                    swp2 = chart.Series[s].Points[i];
            }

            if(swp1 != null && swp2 != null)
                chart.Series[s].Points.Swap(swp1, swp2);
        }

    }

  }

}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraCharts

Namespace ChartStackedSorting
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			chartControl1.Series.Clear()

			Dim s1, s2, s3 As Series

			s1 = New Series("Serie1", ViewType.StackedBar)
			s2 = New Series("Serie2", ViewType.StackedBar)
			s3 = New Series("Serie3", ViewType.StackedBar)

			Dim r As New Random()

			For i As Integer = 0 To 9
				s1.Points.Add(New SeriesPoint(i, Math.Round(r.NextDouble() * 100)))
				s2.Points.Add(New SeriesPoint(i, Math.Round(r.NextDouble() * 100)))
				s3.Points.Add(New SeriesPoint(i, Math.Round(r.NextDouble() * 100)))
			Next i

			s3.Label.Visible = False
			s2.Label.Visible = s3.Label.Visible
			s1.Label.Visible = s2.Label.Visible

			chartControl1.Series.AddRange(New Series() { s1, s2, s3 })
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim css As New ChartStackedSorting(chartControl1)

			css.SortChartSeries()
		End Sub
	End Class

	Public Class ChartStackedSorting
		Private chart As ChartControl

		Public Sub New(ByVal chart As ChartControl)
			Me.chart = chart
		End Sub

		Public Sub SortChartSeries()
			For i As Integer = 0 To chart.Series(0).Points.Count - 1
				Dim p1 As SeriesPoint = chart.Series(0).Points(i)
				Dim pointMax As SeriesPoint = p1
				Dim indexMax As Integer = i

				Dim secondItemsMax As Double = GetAdditionalValues(i)

				For j As Integer = i + 1 To chart.Series(0).Points.Count - 1
					Dim p2 As SeriesPoint = chart.Series(0).Points(j)

					Dim secondItems As Double = GetAdditionalValues(j)

					If p2.Values(0) + secondItems > pointMax.Values(0) + secondItemsMax Then
						pointMax = p2
						secondItemsMax = secondItems
						indexMax = j
					End If
				Next j

				chart.Series(0).Points.Swap(p1, pointMax)
				SwapAdditionalPoints(i, indexMax)
			Next i
		End Sub

		Private Function GetAdditionalValues(ByVal pointIndex As Integer) As Double
			Dim result As Double = 0.0

			For i As Integer = 1 To chart.Series.Count - 1
				result += chart.Series(i).Points(pointIndex).Values(0)
			Next i

			Return result
		End Function

		Private Sub SwapAdditionalPoints(ByVal i As Integer, ByVal j As Integer)
			For s As Integer = 1 To chart.Series.Count - 1
				chart.Series(s).Points.Swap(chart.Series(s).Points(i), chart.Series(s).Points(j))
			Next s
		End Sub

	End Class

End Namespace